src/PenSkin.js

```diff
@@ -130,13 +201,177 @@ class PenSkin extends Skin {
+                data: [
+                    1, 0.5,
+                    0, 0.5,
+                    1, 0,
+
+                    1, 0,
+                    0, 0.5,
+                    0, 0,
+
+                    0.5, 0.5,
+                    0.5, 0.5,
+                    0.5, 0.5,
+
+                    0.5, 0.5,
+                    0.5, 0.5,
+                    0.5, 0.5,
+
+                    1, 0,
+                    0, 0,
+                    1, 0.5,
+
+                    1, 0.5,
+                    0, 0,
+                    0, 0.5
+                ]
+            }
+        };
+
+        this._lineBufferInfo = twgl.createBufferInfoFromArrays(this._renderer.gl, quads);
+    }
+
+    /**
+     * Draw a line on the framebuffer.
+     * @param {PenAttributes} penAttributes - how the line should be drawn.
+     * @param {number} x0 - the X coordinate of the beginning of the line.
+     * @param {number} y0 - the Y coordinate of the beginning of the line.
+     * @param {number} x1 - the X coordinate of the end of the line.
+     * @param {number} y1 - the Y coordinate of the end of the line.
+     */
+    _drawLineOnBuffer (penAttributes, x0, y0, x1, y1) {
+        const gl = this._renderer.gl;
+
+        const bounds = this._bounds;
+        const projection = twgl.m4.ortho(0, bounds.width, 0, bounds.height, -1, 1, __projectionMatrix);
+
+        const NO_EFFECTS = 0;
+        const currentShader = this._renderer._shaderManager.getShader(ShaderManager.DRAW_MODE.lineSample, NO_EFFECTS);
+
+        this._renderer.enterDrawRegion(this._lineOnBufferDrawRegionId, () => {
+            twgl.bindFramebufferInfo(gl, this._framebuffer);
+
+            // Needs a blend function that blends a destination that starts with
+            // no alpha.
+            gl.blendFuncSeparate(gl.ONE, gl.ONE_MINUS_SRC_ALPHA, gl.SRC_ALPHA, gl.ONE);
+
+            gl.viewport(0, 0, bounds.width, bounds.height);
+
+            gl.useProgram(currentShader.program);
+
+            twgl.setBuffersAndAttributes(gl, currentShader, this._lineBufferInfo);
+
+            const uniforms = {
+                u_skin: this._texture,
+                u_projectionMatrix: projection,
+                u_fudge: 0
+            };
+
+            twgl.setUniforms(currentShader, uniforms);
+        });
+
+        const diameter = penAttributes.diameter || DefaultPenAttributes.diameter;
+        const length = Math.hypot(Math.abs(x1 - x0) - 0.001, Math.abs(y1 - y0) - 0.001);
+        const avgX = (x0 + x1) / 2;
+        const avgY = (y0 + y1) / 2;
+        const theta = Math.atan2(y0 - y1, x0 - x1);
+
+        const translationVector = __modelTranslationVector;
+        translationVector[0] = avgX;
+        translationVector[1] = avgY;
+
+        const scalingVector = __modelScalingVector;
+        scalingVector[0] = diameter;
+        scalingVector[1] = length + diameter;
+
+        const radius = diameter / 2;
+        const yScalar = (0.50001 - (radius / (length + diameter)));
```

Comment:
`0.5` here would collapse the mesh to all y values of 0.

In case the length is `0`, `yScalar` will be `0.00001` and will reduce the large `100000` components to `1`, which get clamped to `0.5`. So the large y components of the mesh are never less than `0.5`. The small `0.00001` is intended to be small enough to not produce a visual side effect on any length 0 lines of any diameter. In other-words it as close as possible it should look like a dot or circle and not a very short line.

Reply:
Ah, got it. Thanks!